Index: new/java/src/com/tigervnc/vncviewer/Makefile
===================================================================
--- new.orig/java/src/com/tigervnc/vncviewer/Makefile	2013-04-20 15:24:59.000000000 +0200
+++ new/java/src/com/tigervnc/vncviewer/Makefile	2013-04-20 15:31:48.000000000 +0200
@@ -4,7 +4,7 @@
 
 CP = cp
 JC = javac
-JCFLAGS = -target 1.5 -classpath ../../../
+JCFLAGS = -target 1.7 -classpath ../../../
 JAR = jar
 ARCHIVE = VncViewer.jar
 MANIFEST = MANIFEST.MF
