Index: new/java/src/com/tigervnc/vncviewer/RfbProto.java
===================================================================
--- new.orig/java/src/com/tigervnc/vncviewer/RfbProto.java	2011-01-20 08:14:34.000000000 +0100
+++ new/java/src/com/tigervnc/vncviewer/RfbProto.java	2011-01-20 08:19:20.000000000 +0100
@@ -938,6 +938,9 @@
 
     int keyChar = evt.getKeyChar();
 
+    int test = evt.getKeyCode();
+    System.out.println("KEYCODE: " + evt.getKeyText(test));
+ 
     //
     // Ignore event if only modifiers were pressed.
     //
