Index: new/java/src/com/tigervnc/vncviewer/MANIFEST.MF
===================================================================
--- new.orig/java/src/com/tigervnc/vncviewer/MANIFEST.MF	2011-08-18 14:05:33.000000000 +0200
+++ new/java/src/com/tigervnc/vncviewer/MANIFEST.MF	2013-11-08 11:17:13.000000000 +0100
@@ -1,2 +1,6 @@
 Manifest-Version: 1.0
 Main-Class: com.tigervnc.vncviewer.VncViewer
+Permissions: all-permissions
+Application-Name: vncterm
+Codebase: *
+Caller-Allowable-Codebase: *
